libraries/rush-lib/src/logic/operations/ShellOperationRunner.ts

```diff
@@ -444,10 +444,17 @@ export class ShellOperationRunner implements IOperationRunner {
               const additionalProjectOutputFilePaths: ReadonlyArray<string> = [
                 OperationStateFile.getFilenameRelativeToProjectRoot(this._phase)
               ];
+              const additionalContext: Record<string, string> = {};
+              if (operationSettings.dependsOnEnvVars) {
+                for (const varName of operationSettings.dependsOnEnvVars) {
+                  additionalContext['$' + varName] = process.env[varName] || '';
```

Comment:
How about:

```
  for (const varName of operationSettings.dependsOnEnvVars) {
    if (varName in process.env) {
      additionalContext['$' + varName] = process.env[varName] || '';
    }
```

If the env var is missing in `process.env`, i would like to keep it missing in `additionContext`.

The use case would be like: `"dependsOnEnvVars": ["FOO"]` produces hash A, later if i change it to `"dependsOnEnvVars": ["FOO", "BAR"]`, it still produce hash A when "BAR" is missing, which can make the pervious hash restored in this case.

Reply:
🤔 This seems reasonable but I'm not sure how consistent it is.

Let's say BAR is unusual, and nobody has it set, then adding BAR "for free" is very appealing.

But what if BAR is actually SDK_HOME that everyone has set; the hash will immediately change _except_ for the one new developer that hasn't had their environment set up properly, who keeps pulling the old cache.

(I guess my instinct is that merely indicating there are additional inputs - ESPECIALLY if one day down the road it also whitelists the vars passed down to the operation script.)

Reply:
> the hash will immediately change except for the one new developer that hasn't had their environment set up properly, who keeps pulling the old cache.

In this case, the old cache would be not exist at all 😉. Since remote build cache is only written allowed by a trusted machine(not by a "new developer" environment)

Reply:
Right... hmm.

Maybe I'm coming around on this. I was comparing this in my head to the list of output folder names... which, even if they aren't actually written, just _updating_ the list does affect the generated hash key (it's treated as a "configuration change" for the operation).

But maybe your list of environment variables is more akin to your list of possible command line options. If I go to my `command-line.json` and add several new optional boolean flags for the `build` phase, it only affects my cache hash if I actually _add one of those boolean flags_ to the command line -- the mere possibility of additional flags does not change the hash.